test/integration/persistence.test.ts

```diff
@@ -286,21 +334,48 @@ describe("Model persistence", () => {
       beforeEach(() => {
         fetchMock.restore()
 
-        fetchMock.mock({
-          matcher: "http://example.com/api/v1/people/1",
-          response: new Response({ status: 202 } as any)
-        })
+        fetchMock.delete("http://example.com/api/v1/people/1", { status: 202 })
       })
 
       afterEach(() => {
         resetMocks()
       })
 
       it("does not blow up", async () => {
+        expect(instance.isPersisted).to.eq(true)
+        const success = await instance.destroy()
+
+        expect(success).to.be.true
+        expect(instance.isPersisted).to.eq(true)
+      })
+    })
+
+    describe("when the server returns 202 and a background job object", () => {
+      beforeEach(() => {
+        fetchMock.restore()
+
+        fetchMock.delete("http://example.com/api/v1/people/1", {
+          status: 202,
+          body: { data: { id: "23", type: "background_jobs", attributes: {} } }
+        })
+      })
+
+      afterEach(() => {
+        resetMocks()
+      })
+
+      it("deserialize the response and return the job object", async () => {
+        let job: BackgroundJob = new BackgroundJob()
+        let populateJobOnResponse = (response: any) => {
+          job = response
+        }
+        instance.onDeferredDestroy = populateJobOnResponse
         expect(instance.isPersisted).to.eq(true)
         await instance.destroy()
 
-        expect(instance.isPersisted).to.eq(false)
+        expect(instance.isPersisted).to.eq(true)
+        expect(job.id).to.eq("23")
+        expect(job.klass).to.eq(BackgroundJob)
```

Comment:
Is the same test here twice?

Also would you mind changing test description to "...and calls onDeferredDestroy"?

Reply:
I don't think it's the same test twice. One test is for `save()` and one test is for `destroy()`.